LaunchDarkly/LaunchDarkly/ServiceObjects/Cache/FeatureFlagCache.swift

```diff
@@ -57,18 +57,24 @@ protocol FeatureFlagCaching {
     func saveCachedData(_ storedItems: StoredItems, cacheKey: String, contextHash: String, lastUpdated: Date, etag: String?)
 }
 
+extension MobileKey {
+    func cacheKey() -> String {
```

Comment:
I didn't consider this as an important thing to add (trying to make the PR smaller), but this certainly can be done. Ideally, a wider refactoring is needed here. The parameter name `cacheKey` is confusing because it means different entities in different contexts, while always being a `String`. This cost me a couple of hours of painful debugging to realize that:

* `makeKeyedValueCache(cacheKey: String?)`
* `func getCachedData(cacheKey: String, ...)` and `func saveCachedData(_ storedItems: StoredItems, cacheKey: String, ...)`

use cacheKey that look different, and are generated differently. The first one usually looks like `com.launchdarkly.client....`, and the second one is `context.fullyQualifiedHashedKey()`

Reply:
That's fair. It may be something we add ourselves then after this lands. 

@tanderson-ld that is something you and I can decide on then if we want this now, later, or not at all.